### PR TITLE
EFS - Grant 'endpoints' access

### PR DESCRIPTION
**Pull request by kppullin**
#### What this PR does / why we need it:

The EFS provisioner process requires access to the `endpoints` k8s resource.  This PR adds permissions to `endpoints` to the generated ClusterRole.

You'll see this error message in the logs using the current ClusterRole template:

```
E1108 19:59:11.054882       1 leaderelection.go:238] error initially creating leader election record: endpoints is forbidden: User "system:serviceaccount:kube-system:efs-provisioner" cannot create endpoints in the namespace "kube-system"
```

#### Which issue this PR fixes

Fixes this issue: https://github.com/kubernetes-incubator/external-storage/issues/953

#### Special notes for your reviewer:

/CC @whereisaaron 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [NA] Variables are documented in the README.md

### CODE REVIEW

File: stable/efs-provisioner/Chart.yaml
Comment (whereisaaron) [inline stable/efs-provisioner/Chart.yaml]:
```suggestion
appVersion: v2.1.0-k8s1.11
```

Comment (whereisaaron) [reply stable/efs-provisioner/Chart.yaml]: Hi @kppullin, I suggest you update the `appVersion` to the latest version you have tested with.
https://quay.io/repository/external_storage/efs-provisioner

**Review by whereisaaron — APPROVED** [review]
This extra RBAC permission doesn't break anything for older versions of `efs-provisioner` or Kubernetes, so it seems like a safe change. The change itself looks correct as it matches the [change made in the `efs-provisioner` project](https://github.com/kubernetes-incubator/external-storage/commit/bcdfde83a964ddf238ec2d57a9699d593645cadb) sample manifest.

**Comment by kppullin** [comment]
Thanks for the review and comments @whereisaaron.  I've updated the EFS version info to `v2.0.0-k8s1.11` as suggested, which is what I've tested with and validated as working in our development cluster.

I'm having trouble signing off on the commit generated by your suggested change... do you need to sign off on that or should I keep trying on my end?

**Comment by whereisaaron** [comment]
Oh dear @kppullin. I can't signoff the suggested change or even delete the suggested change since I'd need 'push' access for that. 'Suggested changes' is a new github feature, possibly it isn't compatible with the DCO tool. Sorry I used it since I can't seem to do anything now. It looks like the suggested change is in your branch. If so, maybe you can delete my commit and just do that change yourself?

**Comment by kppullin** [comment]
@whereisaaron - I was able to rebase & squash all the commits and it looks like the DCO gate has been resolved.

**Comment by kppullin** [comment]
@whereisaaron - Any thoughts on the test failure?  I'm not too familiar with the helm testing process, but it looks like the tests may run on GCE and not AWS?  And even if they run on AWS is there a valid EFS resource provisioned?  I'm guessing the `fs-12345678` referenced in the test output is just a placeholder & example and not expected to be used for testing.

**Comment by whereisaaron** [comment]
I don't know @kppullin. This chart has a necessary external dependency the testing doesn't provide. So I imagine there is no way that test can pass. Again I'd have to defer to @davidkarlsen who may know how to skip/ignore that test,

**Comment by kppullin** [comment]
@davidkarlsen - Any thoughts on bypassing the test here? 

Best I can tell the tests run on GCP and not AWS.  Without having an AWS account with an EFS volume to test against there seems little value in the e2e testing of this chart : /

If needed it looks like we make some minor tweaks to bypass creating the `deployment` so that the tests pass, similar to what the `cluster-autoscaler` chart appears to do.

**Comment by kppullin** [comment]
I went ahead and updated the chart to skip creating the deployment when running in CI, as to work around not having a valid AWS EFS resource to test against.  (as mentioned above, this appears to be similar to how the `cluster-autoscaler` chart works around similar limitations).

**Comment by whereisaaron** [comment]
https://github.com/kubernetes-incubator/external-storage/issues/953#issuecomment-449519737
> Same as the two above comments—after applying the additional rules in the PR (https://github.com/helm/charts/pull/9127/files#diff-1f3ae64e932358240df168628073a894R25), I was able to start binding mounts. Only 3 hours wasted here, but quite annoying nonetheless.

**Comment by kppullin** [comment]
Bumping. @davidkarlsen can you take a look or suggest another reviewer?

**Comment by kppullin** [comment]
@lachie83 - do you have a moment to take a look at this PR?  (I noticed you recently merged in another EFS provisioner update with PR #10278).

**Review by lachie83 — APPROVED** [review]

**Comment by kppullin** [comment]
@whereisaaron - Can you give an approval on this change so Lachie83 can move forward with a merge?

**Review by whereisaaron — APPROVED** [review]

**Comment by kppullin** [comment]
@lachie83 - can we get a LGTM on this, now that's it is approved?

**Comment by kppullin** [comment]
@lachie83 bump : )

**Pull request merged**
